app/main.py

```diff
@@ -1,34 +1,106 @@
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.decks = []
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        for row in range(start[0], end[0] + 1):
+            for column in range(start[1], end[1] + 1):
+                self.decks.append(Deck(row, column))
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck | None:
+        for deck in self.decks:
+            if [deck.row, deck.column] == [row, column]:
+                return deck
+        return None
+
+    def fire(self, row: int, column: int) -> None:
+        deck = self.get_deck(row, column)
+        if deck:
+            deck.is_alive = False
+            if True not in [deck.is_alive for deck in self.decks]:
+                self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[tuple[tuple[int, int]]]) -> None:
+        self.field = {
+            ship: Ship(ship[0], ship[1])
+            for ship in ships
+        }
+
+    def fire(self, location: tuple) -> str:
+        for ship in self.field:
+            if self.field[ship].get_deck(location[0], location[1]):
+                self.field[ship].fire(location[0], location[1])
+                if self.field[ship].is_drowned:
+                    return "Sunk!"
+                return "Hit!"
+        return "Miss!"
+
+    def print_field(self) -> None:
+        field = [["~"] * 10 for _ in range(10)]
+        for ship in self.field.values():
+            for deck in ship.decks:
+                if not deck.is_alive:
+                    symbol = "x" if ship.is_drowned else "*"
+                else:
+                    symbol = u"\u25A1"
+                field[deck.row][deck.column] = symbol
+        for row in field:
+            print(" ".join(row))
```

Comment:
The `print_field` method uses a fixed field size of 10x10. If the game rules allow for different field sizes, this should be parameterized.